components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractKeyManager.java

```diff
@@ -135,7 +129,13 @@ public OAuthApplicationInfo buildFromJSON(OAuthApplicationInfo oAuthApplicationI
             if (jsonObject != null) {
                 //create a map to hold json parsed objects.
                 Map<String, Object> params = (Map) jsonObject;
-
+                String grantTypeString = params.get(APIConstants.JSON_GRANT_TYPES).toString();
+                if (StringUtils.isEmpty(oAuthApplicationInfo.getCallBackURL()) &&
+                        StringUtils.isNotEmpty(jsonInput) &&
```

Comment:
redundant check StringUtils.isNotEmpty(jsonInput)

Reply:
Fixed in 801258507d576ca22286f205443071aa999e20e0